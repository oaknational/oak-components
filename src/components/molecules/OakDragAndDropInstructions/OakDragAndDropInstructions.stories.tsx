import React from "react";
import { Meta, StoryObj } from "@storybook/react";

import { OakDragAndDropInstructions } from "./OakDragAndDropInstructions";

const meta: Meta<typeof OakDragAndDropInstructions> = {
  component: OakDragAndDropInstructions,
  tags: ["autodocs"],
  title: "components/molecules/OakDragAndDropInstructions",
  parameters: {
    controls: {
      include: [],
    },
    backgrounds: {
      default: "light",
    },
  },
  render: (args) => <OakDragAndDropInstructions {...args} />,
};
export default meta;

type Story = StoryObj<typeof OakDragAndDropInstructions>;

export const Default: Story = {};
