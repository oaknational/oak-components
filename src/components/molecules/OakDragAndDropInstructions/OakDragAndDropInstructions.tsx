import React, { ComponentPropsWithoutRef } from "react";
import styled from "styled-components";

import { OakBox, OakFlex, OakIcon, OakKbd } from "@/components/atoms";
import { parseSpacing } from "@/styles/helpers/parseSpacing";

/**
 * Adds additional leading between each line of text to make room for the
 * keyboard instructions wrapped in `<OakKbd>`
 *
 * This might be a useful atom to extract
 */
const StyledLeadingTrim = styled(OakBox)`
  margin-block: calc(-${parseSpacing("spacing-8")} / 2);
  line-height: calc(1.5rem + ${parseSpacing("spacing-8")});
`;

/**
 * Displays instructions for drag and drop functionality
 */
export const OakDragAndDropInstructions = (
  props: ComponentPropsWithoutRef<typeof OakFlex>,
) => {
  return (
    <OakFlex $gap="spacing-8" {...props}>
      <OakFlex $flexGrow={0}>
        <OakIcon iconName="move-arrows" />
      </OakFlex>
      <StyledLeadingTrim $font="body-2">
        Click and drag answers to change the order, or select using{" "}
        <OakKbd>
          <span aria-hidden="true">↹</span> Tab
        </OakKbd>{" "}
        then move by pressing <OakKbd>Space</OakKbd> and the <OakKbd>↑</OakKbd>{" "}
        <OakKbd>↓</OakKbd> arrows on your keyboard.
      </StyledLeadingTrim>
    </OakFlex>
  );
};
