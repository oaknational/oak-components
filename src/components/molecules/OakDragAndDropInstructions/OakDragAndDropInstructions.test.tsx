import React from "react";
import "@testing-library/jest-dom";

import { OakDragAndDropInstructions } from "./OakDragAndDropInstructions";

import renderWithTheme from "@/test-helpers/renderWithTheme";

describe(OakDragAndDropInstructions, () => {
  it("matches snapshot", () => {
    const { container } = renderWithTheme(<OakDragAndDropInstructions />);

    expect(container).toMatchSnapshot();
  });
});
