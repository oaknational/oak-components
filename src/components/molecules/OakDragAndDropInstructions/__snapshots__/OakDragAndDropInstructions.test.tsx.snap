// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakDragAndDropInstructions matches snapshot 1`] = `
.c0 {
  font-family: Lexend,sans-serif;
}

.c3 {
  position: relative;
  width: 2rem;
  min-width: 2rem;
  height: 2rem;
  min-height: 2rem;
  font-family: Lexend,sans-serif;
}

.c5 {
  font-family: Lexend,sans-serif;
  font-weight: 300;
  font-size: 1rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
}

.c4 {
  object-fit: contain;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

.c7 {
  background: #ffffff;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  border: 0.125rem solid;
  border-color: #7c9aec;
  border-radius: 0.375rem;
  font-family: Lexend,sans-serif;
  font-weight: 700;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
  white-space: nowrap;
}

.c6 {
  margin-block: calc(-0.5rem / 2);
  line-height: calc(1.5rem + 0.5rem);
}

<div
  className="c0 c1"
>
  <div
    className="c0 c2"
  >
    <div
      className="c3"
    >
      <img
        alt="move-arrows"
        className="c4"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1709052869/icons/hoddjsgpzkszgvnmn91q.svg"
        style={
          {
            "bottom": 0,
            "color": "transparent",
            "height": "100%",
            "left": 0,
            "objectFit": undefined,
            "objectPosition": undefined,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div
    className="c5 c6"
  >
    Click and drag answers to change the order, or select using
     
    <kbd
      className="c7"
    >
      <span
        aria-hidden="true"
      >
        ↹
      </span>
       Tab
    </kbd>
     
    then move by pressing 
    <kbd
      className="c7"
    >
      Space
    </kbd>
     and the 
    <kbd
      className="c7"
    >
      ↑
    </kbd>
     
    <kbd
      className="c7"
    >
      ↓
    </kbd>
     arrows on your keyboard.
  </div>
</div>
`;
